### ISSUE TITLE
Version aliases

### ISSUE THREAD

**Issue by fulldecent** [rbenv/ruby-build#2165]
Please make this work:

`rbenv install 2`

this and 2.7 should currently install 2.7.7.

**Comment by eggplants** [comment]
`pyenv` has already implemented this feature.

ref: https://github.com/pyenv/pyenv/pull/2487

**Comment by mmu094** [comment]
Restart

**Comment by mmu094** [comment]
> Please make this work:
> 
> `rbenv install 2`
> 
> this and 2.7 should currently install 2.7.7.

Is it all OK now

**Comment by jasonkarns** [comment]
Duplicate of https://github.com/rbenv/ruby-build/issues/261 which has been [considered and rejected](https://github.com/rbenv/ruby-build/issues/261#issuecomment-26946265). 

Extra questions that must be answered were this to be considered:
In a scenario where 2.7.0 is already installed, and latest available is 2.7.1, what should `rbenv install 2` do?

1. nothing; the requested version is already satisfied with an already-installed version. surprises users that it does not install the latest available
2. install 2.7.1; user gets latest released version, but surprises users that already have a satisfying version installed. Slightly more concerning, running the same command on the same machine on different days results in different behavior.

**Comment by fulldecent** [comment]
If it is not acceptable to have fuzzy matching to install/use the latest version, then a useful behavior can be that a fuzzy match will result in a message

"The latest available version of ruby 2 is ruby 2.1.99 you can install this with

Rbenv install 2.1.99"

**Comment by mislav** [comment]
> Duplicate of #261 which has been [considered and rejected](https://github.com/rbenv/ruby-build/issues/261#issuecomment-26946265).

@jasonkarns True, but I'm willing to reconsider it. Sometimes I'm lazy and I'd like for ruby-build to just install me the latest version of Ruby 3, whichever that is. Note that if this ships, fuzzy version matching would only work when installing, but _not_ when activating a Ruby version through rbenv.

> Extra questions that must be answered were this to be considered: In a scenario where 2.7.0 is already installed, and latest available is 2.7.1, what should `rbenv install 2` do?

That's a good question! I'd vote for installing the latest available matching version, unless that exact version is already installed.

> If it is not acceptable to have fuzzy matching to install/use the latest version, then a useful behavior can be that a fuzzy match will result in a message

@fulldecent That is also a good idea; thanks!

Let me think about this for a while.

**Comment by jasonkarns** [comment]
FWIW, I've poked at this with nodenv a bit. The issue we've run into is how tightly ruby-build couples the build definition file and the resulting built ruby. So what I had spiked out a few times was to add a generic function (to be used by the build def) to node-build that renames a built version according to its actual full version name. Then we could have a build definition just named `3` that operated similarly to the `-edge` builds (dynamically choose latest download, then rename to exact version after building).

We kept hitting gotchas, but nothing seemed horribly broken by the approach.

**Comment by jasonkarns** [comment]
@eggplants what does pyenv do in the scenario where a user requests a fuzzy version that is already satisfied by an installed version? (say: `pyenv install 3.14` when 3.14.0 is already installed, but 3.14.5 is the latest available?)

Does it behave as `pyenv install 3.14.0`, since 3.14 is already satisfied? (exit with "already installed" message) Or behave as `pyenv install 3.14.5`, since that's the latest available? (actually install latest) Something else that is a blend of the two with custom messaging?
